extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeChannelExtractor.java

```diff
@@ -62,6 +64,29 @@ public void onFetchPage(@Nonnull Downloader downloader) throws IOException, Extr
         String channelUrl = super.getUrl() + CHANNEL_URL_PARAMETERS;
         String pageContent = downloader.download(channelUrl);
         doc = Jsoup.parse(pageContent, channelUrl);
+
+        String feedUrl = getFeedUrl();
+        retrieveVideoTimestampFromXml(Jsoup.parse(
+                downloader.download(feedUrl),
+                feedUrl,
+                org.jsoup.parser.Parser.xmlParser()));
```

Comment:
Content should not be extracted in the `onFetchPage` method. This method should only fetch pages, so you should save the result of `Jsoup.parse(downloader.download(feedUrl), feedUrl, org.jsoup.parser.Parser.xmlParser())` in a variable. This way data can be extracted **on demand**, making the extractor faster when not all info is needed. So you should call `retrieveVideoTimestampFromXml` in `getPublishIsoTimeStrLookup` with the saved variable from above.